cmd/data-crawler.go

```diff
@@ -451,6 +451,16 @@ func (f *folderScanner) scanQueuedLevels(ctx context.Context, folders []cachedFo
 			continue
 		}
 
+		healState, err := getAggregatedBackgroundHealState(ctx, false)
```

Comment:
I would rather do this on a per disk level, so we never choose any disks that are currently being full healed.

They should be filtered out here: https://github.com/minio/minio/blob/c097ce9c324fad06a57ceaf265311ef7b5519b57/cmd/erasure.go#L261

Reply:
the disk doesn't have the healing view, this is a top-level logic @klauspost - here getAggregatedBackgroundHealState() talks to neighboring nodes as well to ensure that we don't initiate new disk activity by triggering heal. 

getAggregatedBackgroundHealState() returns a global view of drives being healed, if no drives are being healed crawler can proceed with its logic. 

that is the reason I added this, disk by itself has no global view.

Reply:
Here are the main reasons we shouldn't use a global state:

1) We should never crawl a disk that is being healed in the first place. It will cause huge fluctuations and unneeded re-scans.
2) We should only ever care about disks in our own set.
3) The bloom filter is much more likely to incorrectly show partially healed paths as not updated.

Reply:
Okay since crawl is per drive you want it to be per drive state?